core/arch/arm/sm/psci.c

```diff
@@ -149,9 +150,7 @@ void tee_psci_handler(struct thread_smc_args *args)
 			;
 		break;
 	case PSCI_SYSTEM_RESET:
-		psci_system_off();
-		while (1)
-			;
+		args->a0 = psci_system_reset();
```

Comment:
Please add a comment explaining that we're not following the specification as we're returning and perhaps what normal world can do about that.